charmstore/client.go

```diff
@@ -35,8 +36,9 @@ type BaseClient interface {
 	// charmrepo/csclient/params, but we don't have one yet.
 
 	// LatestRevisions returns the latest revision for each of the
-	// identified charms. The revisions in the provided URLs are ignored.
-	LatestRevisions([]*charm.URL) ([]charmrepo.CharmRevision, error)
+	// identified charms in the given channel. The revisions in the provided
+	// URLs are ignored.
+	LatestRevisions(urls []*charm.URL, channel string) ([]charmrepo.CharmRevision, error)
```

Comment:
As mentioned, the channels are not homogeneous.  So the channel must be associated with each element.